packages/neon/neon_dashboard/lib/src/widgets/widget.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:neon/theme.dart';
+import 'package:neon/widgets.dart';
+import 'package:neon_dashboard/l10n/localizations.dart';
+import 'package:neon_dashboard/src/widgets/widget_item.dart';
+import 'package:nextcloud/dashboard.dart' as dashboard;
+
+class DashboardWidget extends StatelessWidget {
+  const DashboardWidget({
+    required this.widget,
+    required this.items,
+    super.key,
+  });
+
+  final dashboard.Widget widget;
+  final dashboard.WidgetItems? items;
+
+  @override
+  Widget build(final BuildContext context) {
+    final halfEmptyContentMessage = items?.halfEmptyContentMessage ?? '';
```

Comment:
another bad pattern to avoid dart null ability.

Reply:
This is on purpose because the halfEmptyContentMessage is an empty string if is "empty". This way we only need to check if the string is empty or not instead of first checking if the items are null and then checking if the string is empty.

Reply:
Before we go back and for even more here is what I envisioned:
IMO this is more readable than the current approach and better than the old version.

```dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:neon/theme.dart';
import 'package:neon/widgets.dart';
import 'package:neon_dashboard/l10n/localizations.dart';
import 'package:neon_dashboard/src/widgets/widget_button.dart';
import 'package:neon_dashboard/src/widgets/widget_item.dart';
import 'package:nextcloud/dashboard.dart' as dashboard;

class DashboardWidget extends StatelessWidget {
  const DashboardWidget({
    required this.widget,
    required this.items,
    super.key,
  });

  final dashboard.Widget widget;
  final dashboard.WidgetItems? items;

  @override
  Widget build(final BuildContext context) {
    final halfEmptyContentMessage = _renderMessage(items?.halfEmptyContentMessage);
    final emptyContentMessage = _renderMessage(items?.emptyContentMessage);

    return Card(
      child: InkWell(
        onTap: widget.widgetUrl != null && widget.widgetUrl!.isNotEmpty ? () => context.go(widget.widgetUrl!) : null,
        borderRadius: const BorderRadius.all(Radius.circular(12)),
        child: ListView(
          padding: const EdgeInsets.all(8),
          children: [
            ListTile(
              title: Text(
                widget.title,
                style: const TextStyle(
                  fontWeight: FontWeight.bold,
                ),
              ),
              leading: SizedBox.square(
                dimension: largeIconSize,
                child: NeonCachedImage.url(
                  url: widget.iconUrl,
                  svgColorFilter: ColorFilter.mode(Theme.of(context).colorScheme.primary, BlendMode.srcIn),
                  size: const Size.square(largeIconSize),
                ),
              ),
            ),
            const SizedBox(
              height: 20,
            ),
            if (halfEmptyContentMessage != null) halfEmptyContentMessage,
            if (emptyContentMessage != null) emptyContentMessage,
            if (halfEmptyContentMessage == null && emptyContentMessage == null)
              _renderMessage(AppLocalizations.of(context).noEntries)!,
            ...?items?.items.map(
              (final item) => DashboardWidgetItem(
                item: item,
                roundIcon: widget.itemIconsRound,
              ),
            ),
            const SizedBox(
              height: 20,
            ),
            ...?widget.buttons?.map(
              (final button) => DashboardWidgetButton(
                button: button,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget? _renderMessage(final String? message) {
    if (message == null || message.isEmpty) {
      return null;
    }

    return Center(
      child: Column(
        children: [
          const Icon(
            Icons.check,
            size: largeIconSize,
          ),
          Text(message),
        ],
      ),
    );
  }
}

```